lib/blacklight/access_controls/enforcement.rb

```diff
@@ -25,72 +29,61 @@ module Enforcement
+        solr_access_filters_logic.map { |method| send(method, permission_types, ability).reject(&:blank?) }
       end
 
-      # Controller before filter that sets up access-controlled lucene query in order to provide gated discovery behavior
-      # @param solr_parameters the current solr parameters
+      ### Solr query modifications
+
+      # Controller before_filter that sets up access-controlled lucene query to provide gated discovery behavior.
+      # Set solr_parameters to enforce appropriate permissions.
+      # @param [Hash{Object}] solr_parameters the current solr parameters, to be modified herein!
+      # @note Applies a lucene filter query to the solr :fq parameter for gated discovery.
       def apply_gated_discovery(solr_parameters)
         solr_parameters[:fq] ||= []
-        solr_parameters[:fq] << gated_discovery_filters.join(' OR ')
+        solr_parameters[:fq] << gated_discovery_filters.reject(&:blank?).join(' OR ')
         Rails.logger.debug("Solr parameters: #{solr_parameters.inspect}")
       end
 
+      # For groups
+      # @return [Array{String}] values are lucence syntax term queries suitable for :fq
+      # @example
+      #   [ "({!terms f=discover_access_group_ssim}public,faculty,africana-faculty,registered)",
+      #     "({!terms f=read_access_group_ssim}public,faculty,africana-faculty,registered)" ]
       def apply_group_permissions(permission_types, ability = current_ability)
-        # for groups
+        groups = ability.user_groups
+        return [] if groups.empty?
         permission_types.map do |type|
           field = solr_field_for(type, 'group')
-          groups = ability.user_groups
-          # The parens are required to properly OR the cases together.
-          "({!terms f=#{field}}#{groups.join(',')})"
+          "({!terms f=#{field}}#{groups.join(',')})" # parens required to properly OR the clauses together.
         end
       end
 
+      # For individual user access
+      # @return [Array{String}] values are lucence syntax term queries suitable for :fq
+      # @example ['discover_access_person_ssim:user_1@abc.com', 'read_access_person_ssim:user_1@abc.com']
       def apply_user_permissions(permission_types, ability = current_ability)
-        # for individual user access
-        user_access_filters = []
         user = ability.current_user
-        if user && user.user_key.present?
-          permission_types.each do |type|
-            user_access_filters << escape_filter(solr_field_for(type, 'user'), user.user_key)
-          end
+        return [] unless user && user.user_key.present?
+        permission_types.map do |type|
+          escape_filter(solr_field_for(type, 'user'), user.user_key)
         end
-        user_access_filters
       end
 
-      # Find the name of the solr field for this type of permission.
-      # e.g. "read_access_group_ssim" or "discover_access_person_ssim".
+      # @param [#to_s] permission_type, a single value, e.g. "read" or "discover"
```

Comment:
Don't use a comma after the param name see http://www.rubydoc.info/gems/yard/file/docs/Tags.md#param